lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -266,6 +266,12 @@ def install_puppet_on(hosts, opts={})
               if host['platform'] =~ /el-(5|6|7)/
                 relver = $1
                 install_puppet_from_rpm_on(host, opts.merge(:release => relver, :family => 'el'))
+              elsif host['platform'] =~ /sles-(11|12)/
+                relver = $1
+                # Need to get rid of whatever Puppet might have shipped on the
+                # upstream images
+                host.uninstall_package('puppet')
```

Comment:
@kevpl the vmpooler sles templates do not have puppet applications installed. As an example, here is the output from a zypper query for puppet on sles 11. It shows that the packages are available in the repos, but are not installed (lack of "i" in the first column).

```
# zypper search -s | grep puppet
  | puppet                                     | srcpackage | 2.6.18-0.4.2                  | noarch | oy.os
  | puppet                                     | package    | 2.6.18-0.4.2                  | x86_64 | oy.os
  | puppet-server                              | package    | 2.6.18-0.4.2                  | x86_64 | oy.os
  | rubygem-kiwi2puppet                        | package    | 0.1.0-0.8.2                   | x86_64 | oy.os

```